File: src/F#-Samples/FSApp/MauiProgram.fs
namespace FSApp

open Microsoft.Extensions.DependencyInjection
open Microsoft.Maui.Accessibility
open Microsoft.Maui.Hosting
open Fabulous.Maui
open VijayAnand.MauiToolkit
open FSApp.Extensions

type MauiProgram =
    static member CreateMauiApp() =
        MauiApp
            .CreateBuilder()
            .UseFabulousApp(App.program)
            .ConfigureFonts(fun fonts ->
                fonts
                    .AddFont("OpenSans-Regular.ttf", "OpenSansRegular")
                    .AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold")
                |> ignore)
            .ConfigureServices(fun services ->
                services
                    .AddSingleton(SemanticScreenReader.Default)
                |> ignore)
#if DEBUG
            .AddDebugLog()
#endif
            .Build()

File: src/NET_10/MauiAspire/MauiApp1/MauiApp1/MauiProgram.cs
using MauiApp1.Services;
using Microsoft.Extensions.Hosting;
using Microsoft.Extensions.Logging;
using VijayAnand.MauiToolkit.Core.Services;

namespace MauiApp1
{
    public static partial class MauiProgram
    {
        public static MauiApp CreateMauiApp()
        {
            var builder = MauiApp.CreateBuilder();
            builder.UseMauiApp<App, MainWindow, MainPage>()
                   .AddServiceDefaults() // Aspire service defaults
                   .ConfigureFonts(fonts =>
                   {
                       fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
                       fonts.AddFont("OpenSans-SemiBold.ttf", "OpenSansSemiBold");
                   })
                   .ConfigureServices(services =>
                   {
                       services.AddSingleton<MainViewModel>();
                       services.AddSingleton<WeatherDataStore>();
                       // BaseAddress should point to the named endpoint registered in the Aspire project.
                       // It'll be auto resolved using Aspire's service discovery.
                       services.AddHttpClient("webapi", client => client.BaseAddress = new("https+http://webapi"));
                   });

#if DEBUG
            builder.Logging.AddDebug();
#endif

            return builder.Build();
        }
    }
}
